### DISCUSSION TITLE
Defer highlighting for completion styles

### DISCUSSION THREAD

**Discussion by kiennq** [minad/corfu#441] [Q&A] [answered]
Hi @minad, you have mentioned about the defer highlighting in this comment https://github.com/emacs-lsp/lsp-mode/pull/2975#discussion_r662028340.
Can you please share more information about that? Is there a propose on upstream for that already?

**Comment by minad** [comment, marked as answer]
Deferred highlighting is implemented upstream, and supported for example by the Orderless completion style. See the variables `completion-lazy-hilit` and `completion-lazy-hilit-fn`. I don't think this feature is useful for lsp-mode, which wants to apply its own highlighting to the candidates?

**Reply by kiennq** [reply]
Thanks, `lsp-mode` doesn't do its own highlighting but instead provides hints to external package (`:company-match` for `company-node`) to have the candidates highlighted.
The `completion-lazy-hilit-fn` is perfect, I can just set the `:company-match` function to it.
This is available from Emacs 30 right?

**Reply by minad** [reply]
> The completion-lazy-hilit-fn is perfect, I can just set the :company-match function to it.
This is available from Emacs 30 right?

Yes, this can work if the completion style doesn't override it again, or if you do it directly in your lsp-mode specific passthrough completion style.

**Reply by kiennq** [reply]
Create a PR here for lsp-mode https://github.com/emacs-lsp/lsp-mode/pull/4394
